llvm/lib/SYCLLowerIR/SYCLUtils.cpp

```diff
@@ -43,6 +47,13 @@ void traverseCallgraphUp(llvm::Function *F, CallGraphNodeAction ActionF,
         } else {
           // ... non-call is OK - add using function to the worklist
           if (auto *I = dyn_cast<Instruction>(FCall)) {
+            if (auto *SI = dyn_cast<StoreInst>(I)) {
```

Comment:
please add comment why this special branch is needed `if (auto *SI = dyn_cast<StoreInst>(I))`